scripts/iso_mirna_quantification.py

```diff
@@ -0,0 +1,337 @@
+`--collapsed` is set, the contribution is # of reads/NH_tag. If only `--nh` is
+set, the contribution is 1/NH. Otherwise, the contribution is 1/NH_tag. If
+`--nh` is not set and the NH tag is not found, the value is set to 1. If the
+SAM file is empty, an empty file is produced. The output columns are also
+determined by the flags `--read-ids` and `--len`. If `--len` is set, the
+length of the species sequence is add as a new column in the output table.
+If `--read-ids` is set, a semicolon-separated list of all alignment IDs that
+overlap with the feature will always be in the last column. The output file(s)
+are determined by the arguments `--mir-list` and `--lib`. The `--mir-list`
+determines which tables are created and how the classification is done. If the
+value of `--mir-list` is 'iso_mirna', a single table is created that includes
+both isomiR and canonical miRNAs. If the value is either 'isomir' or 'mirna',
+a single table is created containing only the specified feature type. If the
+value is 'isomir' and 'mirna', two separate tables are created for each feature
+type. The `--lib` argument determines the suffix of the output table. If it is
+not specified, the suffix 'lib' is used.
+"""
+
+import argparse
+from pathlib import Path
+from typing import Optional
+
+import pysam
+
+
+def parse_arguments():
+    """Command-line arguments parser."""
+    parser = argparse.ArgumentParser(
+        description=__doc__
+        )
+    parser.add_argument(
+        '-v', '--version',
+        action='version',
+        version='%(prog)s 1.0',
+        help="Show program's version number and exit"
+    )
+    parser.add_argument(
+        'samfile',
+        help=(
+            "Path to the SAM file sorted by a the tag containing read"
+            "intersecting species name."
+        ),
+        type=Path
+    )
+    parser.add_argument(
+        '--outdir',
+        help="Path to the output directory. Default: %(default)s.",
+        default=Path.cwd(),
+        type=Path
+    )
+    parser.add_argument(
+        '--mir-list',
+        help=(
+            "List of miRNA species to have in the different output tables."
+            "Default: %(default)s"
+        ),
+        nargs='*',
+        default=['iso_mirna'],
+        type=str
+    )
+    parser.add_argument(
+        '--lib',
+        help=(
+            "Library to which the alignments belong to. Default : %(deafult)s"
+        ),
+        type=str,
+        default="lib",
+    )
+    parser.add_argument(
+        '-t', '--tag',
+        help=(
+            "Indicate the tag storing the read species name in uppercase."
+            "Default %(default)s."
+        ),
+        default='YW',
+        type=str
+    )
+    parser.add_argument(
+        '-c', '--collapsed',
+        help=(
+            "Indicate that the SAM file has the reads collapsed by sequence"
+            "and alignment. The collapsed name must be build by the alignment"
```

Comment:
By sequence and alignment? What do you mean? I thought it's just the reads collapsed with `fastx-collapser`. You can reference that also here, e.g.:

> ", as, e.g., produced by the 'fastx_collapser' tool of the FASTX-Toolkit: http://hannonlab.cshl.edu/fastx_toolkit/".

To give you one example here for a thorough description (but please reconsider the descriptions for _all_ CLI options):

"Set to enable quantification if reads in SAM file were collapsed. In that case, SAM query names are expected to follow the general format 'NAME-COUNT', where NAME is an arbitrary unique name and COUNT is the number of identical reads that were collapsed. For example, the query name 'my_query_name-123' would indicate that the read represented by this alignment occurred 123 times before collapsing. The required naming format is produced, e.g., by the 'fastx_collapser' tool of the FASTX-Toolkit: http://hannonlab.cshl.edu/fastx_toolkit/"